code/controllers/subsystem/achievements.dm

```diff
@@ -45,3 +47,29 @@ SUBSYSTEM_DEF(achievements)
 		cheevos_to_save += PD.achievements.get_changed_data()
 	
 	SSdbcore.MassInsert(format_table_name("achievements"),cheevos_to_save,duplicate_key = TRUE)
+
+//Update the metadata if any are behind
+/datum/controller/subsystem/achievements/proc/update_metadata()
+	var/list/current_metadata = list()
+	//select metadata here
+	var/datum/DBQuery/Q = SSdbcore.NewQuery("SELECT achievement_key,achievement_version,achievement_type FROM [format_table_name("achievement_metadata")]")
+	if(!Q.Execute(async = TRUE))
+		qdel(Q)
+		return
+	else
+		while(Q.NextRow())
+			var/ac_key = Q.item[1]
+			var/ac_version = text2num(Q.item[2])
+			current_metadata[ac_key] = ac_version
+		qdel(Q)
+
+	var/list/to_update = list()
+	for(var/T in awards)
+		var/datum/award/A = awards[T]
+		if(!A.database_id)
+			continue
+		if(current_metadata[A.database_id] || current_metadata[A.database_id] < A.achievement_version)
```

Comment:
`current_metadata[A.database_id] ||` won't this update all achievements and not just those behind on version?

Reply:
Whoops, meant to be ! there